test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeControllerSpec.scala

```diff
@@ -416,6 +418,322 @@ class RepresenteeControllerSpec
+      val monthKey = "representee.dateOfDeath-month"
+      val yearKey  = "representee.dateOfDeath-year"
+
+      def formData(dateOfDeath: DateOfDeath): Seq[(String, String)] =
+        Seq(
+          dayKey   -> dateOfDeath.value.getDayOfMonth.toString,
+          monthKey -> dateOfDeath.value.getMonthValue.toString,
+          yearKey  -> dateOfDeath.value.getYear.toString
+        )
+
+      behave like redirectToStartBehaviour(() => performAction(Seq.empty))
+
+      behave like nonCapacitorOrPersonalRepBehaviour(() => performAction(Seq.empty))
+
+      "show an error page" when {
+
+        val oldAnswers = sample[CompleteRepresenteeAnswers].copy(dateOfDeath = Some(DateOfDeath(LocalDate.now())))
+        val newDate    = DateOfDeath(LocalDate.now().minusDays(1))
+        val (session, journey, draftReturn) =
+          sessionWithFillingOutReturn(oldAnswers, Left(PersonalRepresentative))
+        val newDraftReturn = draftReturn.copy(
+          representeeAnswers = Some(oldAnswers.copy(dateOfDeath = Some(newDate)))
+        )
+
+        val newJourney = journey.copy(draftReturn = newDraftReturn)
+
+        "there is an error updating the draft return" in {
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(session)
+            mockStoreDraftReturn(
+              newDraftReturn,
+              newJourney.subscribedDetails.cgtReference,
+              newJourney.agentReferenceNumber
+            )(Left(Error("")))
+          }
+
+          checkIsTechnicalErrorPage(performAction(formData(newDate)))
+        }
+
+        "there is an error updating the session" in {
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(session)
+            mockStoreDraftReturn(
+              newDraftReturn,
+              newJourney.subscribedDetails.cgtReference,
+              newJourney.agentReferenceNumber
+            )(Right(()))
+            mockStoreSession(session.copy(journeyStatus = Some(newJourney)))(Left(Error("")))
+          }
+          checkIsTechnicalErrorPage(performAction(formData(newDate)))
+        }
+
+        "the date entered is invalid" in {
+
+          val session = sessionWithFillingOutReturn(
+            IncompleteRepresenteeAnswers.empty.copy(name = Some(sample[IndividualName])),
+            Left(PersonalRepresentative)
+          )._1
+
+          val key = "representee.dateOfDeath"
+          DateErrorScenarios
+            .dateErrorScenarios(key, "")
+            .foreach { scenario =>
+              withClue(s"For date error scenario $scenario: ") {
+                val data = List(
+                  s"$key-day"   -> scenario.dayInput,
+                  s"$key-month" -> scenario.monthInput,
+                  s"$key-year"  -> scenario.yearInput
+                ).collect { case (key, Some(value)) => key -> value }
+
+                testFormError(performAction, "representee.dateOfDeath.title", session)(data, _)
```

Comment:
what does the underscore refer to here? Am I missing where the expected error message key comes from?